File: basic_ros_example/src/service_client.cpp
#include <ros/ros.h>
#include <basic_ros_example/Adder.h>

int main(int argc, char** argv) {
  ros::init(argc, argv, "service_client");
  ros::NodeHandle node;

  ros::ServiceClient adder_srv = node.serviceClient<basic_ros_example::Adder>("adder_service");

  basic_ros_example::AdderRequest request;
  basic_ros_example::AdderResponse response;
  request.val1 = 4.5;
  request.val2 = 1.0;

  adder_srv.waitForExistence();
  bool success = adder_srv.call(request, response);

  if (success) {
    ROS_INFO_STREAM("Service succeeded! Result: " << response.result);
  } else {
    ROS_WARN("Service call failed!");
  }

  return 0;
}

File: basic_ros_example/scripts/basic_ros_example/service_client.py
#! /usr/bin/env python3
import rospy
from basic_ros_example.srv import Adder

rospy.init_node('service_client')

adder_srv = rospy.ServiceProxy('adder_service', Adder)
adder_srv.wait_for_service()

try:
    response = adder_srv(val1=4.5, val2=1.0)
    rospy.loginfo(f'Service succeeded! Result: {response.result}')
except rospy.ServiceException as e:
    rospy.logwarn(f'Service call failed! {e}')